clients/src/main/resources/common/message/README.md

```diff
@@ -79,9 +87,14 @@ There are several primitive field types available.
 
 * "int64": a 64-bit integer.  This takes up 8 bytes on the wire.
 
-* "string": a string.  This must be less than 64kb in size when serialized as UTF-8.  This takes up 2 bytes on the wire, plus the length of the string when serialized to UTF-8.
+* "string": a string.  Strings are serialized as a length followed by the
+  contents as UTF-8.  The contents must be less than 64kb in size.  In
+  non-flexible versions, the string length will always be 2 bytes.  In flexible
+  versions, the length is a variable-length integer. 
```

Comment:
KIP-482 indicates that flexible versions will actually use unsigned variable length numbers, offset by 1 to reserve 0 indicating null. If that's the case, this should change from `variable-length integer`, which implies the old zig-zag varint, to `unsigned variable-length integer offset by 1, with 0 indicating null` (or something like that).

Also, it might be worth mentioning that the numbers are always offset by one, even if the field is non-nullable.

Reply:
I'm going to split the documentation part off into a separate PR, since there are some tricky questions about what should go in what documentation section.  Let's discuss it there.